api/task_delete.go

```diff
@@ -0,0 +1,60 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/consul-terraform-sync/logging"
+)
+
+// deleteTask deletes an existing task and its events. Does not delete
+// if the task is active.
+func (h *taskHandler) deleteTask(w http.ResponseWriter, r *http.Request) {
+	taskName, err := getTaskName(r.URL.Path, taskPath, h.version)
+	logger := logging.FromContext(r.Context()).Named(deleteTaskSubsystemName)
+	if err != nil {
+		logger.Trace("bad request", "error", err)
+		jsonErrorResponse(r.Context(), w, http.StatusBadRequest, err)
+		return
+	}
+
+	if taskName == "" {
+		err := fmt.Errorf("no task name was included in the api request. " +
+			"Deleting a task requires the task name: '/v1/tasks/:name'")
+		logger.Trace("bad request", "error", err)
+		jsonErrorResponse(r.Context(), w, http.StatusBadRequest, err)
+		return
+	}
+	logger.Trace("deleting task", "task", taskName)
```

Comment:
for consistency, maybe call the field task_name since you're not logging the entire task?

Reply:
same comment for other logs with taskName

Reply:
That makes sense, I'll update!